src/portfft/committed_descriptor_impl.hpp

```diff
@@ -242,17 +224,29 @@ class committed_descriptor_impl {
     std::shared_ptr<IdxGlobal> factors_and_scan;
     detail::level level;
     std::size_t length;
+    std::size_t committed_length;
     Idx used_sg_size;
     Idx num_batches_in_l2;
-    Idx num_factors;
+    Idx num_forward_factors;
+    Idx num_backward_factors;
+    bool is_prime;
+    IdxGlobal backward_twiddles_offset;
+    IdxGlobal bluestein_modifiers_offset;
+    IdxGlobal forward_impl_twiddle_offset;
+    IdxGlobal backward_impl_twiddle_offset;
```

Comment:
Are these bluestein-specific?

Reply:
Depends which one you are talking about. 
* `num_forward/backward_factors` - Not strictly, we previously had a member called `num_factors`, which I have now changed to `forward/backward_factors`
* is_prime - Yes
* Offsets - Again not strictly. This is just maintaining which twiddles start from what offset. We used to calculate the offset here [here](https://github.com/codeplaysoftware/portFFT/blob/d5fe215de93e38244278909d5f2da8c5853e566f/src/portfft/dispatcher/global_dispatcher.hpp#L321) but now I am populating that value during the commit time itself, so as to not spend extra time simply calculating offsets during compute

Reply:
We discussed the *single responsiblity principle* in a recent retro. Would it be possible to add these in a way such that each variable has a single well defined use?

Reply:
I do not see how it is breaking the single responsibility principle, each of these variable is responsible for holding exactly one value which has one specific purpose, that is to determine the starting address of some twiddles.

Reply:
Apologies, I misread your prior comment.